lib/client/doc.js

```diff
@@ -866,6 +864,12 @@ Doc.prototype.submitOp = function(component, options, callback) {
     callback = options;
     options = null;
   }
+
+  // If agent has metadata, append on client level so that both client and backend have access to it
+  if(this.connection.agent && this.connection.agent.custom && typeof this.connection.agent.custom.metadata === "object") {
```

Comment:
I'm not sure I understand why we're doing this? The `custom` object is meant to be a generic data blob for clients to use. Setting a magical property here is surprising and may clash with clients' existing usage.

Also `this.connection` is only ever available in setups where the client is running on the same machine as the server (as opposed to, say, a client connected remotely over websocket).